app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -272,10 +273,36 @@ def validate_channel(self, value):
         return value
 
 
+class NimbusDocumentationLinkSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = NimbusDocumentationLink
+        fields = (
+            "title",
+            "link",
+        )
+
+
+class NimbusExperimentDocumentationLinkMixin:
+    def update(self, experiment, data):
+        documentation_links_data = data.pop("documentation_links", None)
+        experiment = super().update(experiment, data)
+        if documentation_links_data is not None:
+            with transaction.atomic():
```

Comment:
Oh yeah this is what I meant the other day sorry I meant to respond in the other PR, I figured `update` should be inside `transaction.atomic` so the whole thing was atomic?  What do you think?

Reply:
I wouldn't think the entire experiment update needs to be in the transaction - the delete-and-add is the main atomic concern here I think? And that doesn't need doing at all if the property isn't in the update data

Reply:
Yeah that's what I was thinking about, you definitely don't want the delete succeeding and the related insertions failing or visa versa, but does it make sense for that whole section to fail but the other update fields to succeed?  Or should the whole update fail?  I would think the whole update should fail, otherwise it's a partial update and some fields are saved and others aren't?  Like if you're not doing the related entity logic, it's just a single transaction to update the fields on the parent model, and if one of those fields fails that whole update is thrown out, I would think preserving that behaviour would be desired?